modules/editorial-metadata/editorial-metadata.php

```diff
@@ -344,6 +391,57 @@ public function delete_editorial_metadata_term( $term_id ) {
 		return $result;
 	}
 
+	 /** Generate the args for registering post meta
+	 *
+	 * @param WP_Term $term The term object
+	 * @return array $args Post meta args
+	 */
+	public function get_postmeta_args( $term ) {
+		$arg_type = '';
+		switch ( $term->type ) {
+			case 'checkbox':
+				$arg_type = 'boolean';
+				break;
+			case 'text':
+				$arg_type = 'string';
+				break;
+		}
+		$args = [
+			'type' => $arg_type,
+			'description' => $term->description,
+			'single' => true,
+			'show_in_rest' => true,
+			'sanitize_callback' => function ( $value ) use ( $arg_type ) {
+				switch ( $arg_type ) {
+					case 'boolean':
+						return boolval( $value );
+					case 'string':
+						return stripslashes( wp_filter_nohtml_kses( trim( $value ) ) );
+				}
+			},
+			'auth_callback' => function () {
+				// ToDo: Review the permissions required to edit metadata
+				return current_user_can( 'edit_posts' ) || current_user_can( 'edit_pages' );
```

Comment:
Does this make sense?

Reply:
I wasn't sure how this worked, so I spent some time in the debugger. The short answer is I think you can remove `auth_callback` entirely as core handles permission checks properly unless you want to check extra permissions.

When editorial metadata is updated, this initially travels through `update_meta_value()` [to the `current_user_can()`](https://github.com/WordPress/WordPress/blob/741a7869fd77a0f979a2116ae87b1413a0e62fb1/wp-includes/rest-api/fields/class-wp-rest-meta-fields.php#L390) function. This eventually falls to [`WP_User::has_cap()`](https://github.com/WordPress/WordPress/blob/741a7869fd77a0f979a2116ae87b1413a0e62fb1/wp-includes/class-wp-user.php#L776) which checks the `"edit_post_meta"` capability with the current post ID. When the post is a page, this is `map_meta_cap()`ed into the `"edit_pages"` capability, and when it's a post it become `"edit_posts"`. If you don't pass a custom auth callback, it seems like WordPress maps to the relevant capability type properly.

Reply:
Beauty, I struggled a bit to dig into that exact `map_meta_cap` call so appreciate you digging that up! I'll take this out 👍🏾